azure_functions_worker/bindings/shared_memory_manager.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+from __future__ import annotations
+import uuid
+from typing import Dict
+from typing import Optional
+from ..logging import logger
+from ..mmap_handler.file_writer import FileWriter
+from ..mmap_handler.file_reader import FileReader
+from ..mmap_handler.file_accessor_factory import FileAccessorFactory
+
+
+class SharedMemoryManager:
+    """
+    Performs all operations related to reading/writing data from/to shared memory.
+    This is used for transferring input/output data of the function from/to the functions host over
+    shared memory as opposed to RPC to improve the rate of data transfer and the function's
+    end-to-end latency.
+    """
+    def __init__(self):
+        # The allocated memory maps are tracked here so that a reference to them is kept open until
+        # they have been used (e.g. if they contain a function's output, it is read by the
+        # functions host).
+        # Having a mapping of the name and the memory map is then later used to close a given
+        # memory map by its name, after it has been used.
+        # key: map_name, val: mmap.mmap
+        self.allocated_mem_maps: Dict[str, mmap.mmap] = {}
```

Comment:
This is a little convoluted, but what if the worker restarts and the mmaps stay there?
Should we try to repopulate them?

Reply:
Ummm... that's a good point. If the worker restarts in case of error, then there is already trouble. 
The reason why I don't want to repopulate this is because if the memory map was allocated by the functions host or a different worker process, then things can get weird. 
So, we need to think on how to avoid such leaks if the worker restarts.
But then, what if the functions host restarts - even that could possibly leave maps around, right?

Reply:
Yes, I'm pretty sure it would be fine but worth giving a though to those situations.